s3/src/main/scala/akka/stream/alpakka/s3/impl/S3Stream.scala

```diff
@@ -569,53 +569,12 @@ import scala.collection.immutable
 
   val atLeastOneByteString = Flow[ByteString].orElse(Source.single(ByteString.empty))
 
-  private def createRequests(
-      s3Location: S3Location,
-      contentType: ContentType,
-      s3Headers: S3Headers,
-      chunkSize: Int,
-      parallelism: Int
-  ): Flow[ByteString, (HttpRequest, (MultipartUpload, Int)), NotUsed] = {
-
-    assert(
-      chunkSize >= MinChunkSize,
-      s"Chunk size must be at least 5 MB = $MinChunkSize bytes (was $chunkSize bytes). See http://docs.aws.amazon.com/AmazonS3/latest/API/mpUploadUploadPart.html"
-    )
-
-    val chunkBufferSize = chunkSize * 2
-
-    // First step of the multi part upload process is made.
-    //  The response is then used to construct the subsequent individual upload part requests
-    val requestInfo: Source[(MultipartUpload, Int), NotUsed] =
-      initiateUpload(s3Location, contentType, s3Headers.headersFor(InitiateMultipartUpload))
-
-    val headers = s3Headers.serverSideEncryption.toIndexedSeq.flatMap(_.headersFor(UploadPart))
-
-    Flow
-      .setup { (mat, attr) =>
-        implicit val conf = resolveSettings(attr, mat.system)
-
-        SplitAfterSize(chunkSize, chunkBufferSize)(atLeastOneByteString)
-          .via(getChunkBuffer(chunkSize, chunkBufferSize)) //creates the chunks
-          .concatSubstreams
-          .zipWith(requestInfo) {
-            case (chunkedPayload, (uploadInfo, chunkIndex)) =>
-              //each of the payload requests are created
-              val partRequest =
-                uploadPartRequest(uploadInfo, chunkIndex, chunkedPayload.data, chunkedPayload.size, headers)
-              (partRequest, (uploadInfo, chunkIndex))
-          }
-          .flatMapConcat { case (req, info) => Signer.signedRequest(req, signingKey).zip(Source.single(info)) }
-      }
-      .mapMaterializedValue(_ => NotUsed)
-  }
-
-  private def getChunkBuffer(chunkSize: Int, bufferSize: Int)(implicit settings: S3Settings) =
+  private def getChunkBuffer(chunkSize: Int, bufferSize: Int, maxRetriesPerChunk: Int)(implicit settings: S3Settings) =
     settings.bufferType match {
       case MemoryBufferType =>
         new MemoryBuffer(bufferSize)
       case d: DiskBufferType =>
-        new DiskBuffer(2, bufferSize, d.path)
+        new DiskBuffer((maxRetriesPerChunk + 1) * 2, bufferSize, d.path)
```

Comment:
This new value would benefit from a proper name or explanation.

Reply:
Done.